lib/std/crypto/sha2.zig

```diff
@@ -1,87 +1,92 @@
+//! Secure Hashing Algorithm 2 (SHA2)
+//!
+//! Published by the National Institue of Standards and Technology (NIST) [1] [2].
+//!
+//! Truncation mitigates length-extension attacks but increases vulnerability to collision
+//! attacks. Collision attacks remain impractical for all types defined here.
+//!
+//! T: original hash function, whose output is simply truncated.
+//!    A truncated output is just the first bytes of a longer output.
+//! _: hash function with context separation.
+//!    Different lengths produce completely different outputs.
+//!
+//! [1] https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf
+//! [2] https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-208.pdf
+
 const std = @import("../std.zig");
 const builtin = @import("builtin");
 const mem = std.mem;
 const math = std.math;
 const htest = @import("test.zig");
 
-/////////////////////
-// Sha224 + Sha256
-
-const RoundParam256 = struct {
-    a: usize,
-    b: usize,
-    c: usize,
-    d: usize,
-    e: usize,
-    f: usize,
-    g: usize,
-    h: usize,
-    i: usize,
+pub const Sha224 = Sha2x32(iv224, 224);
+pub const Sha256 = Sha2x32(iv256, 256);
+pub const Sha384 = Sha2x64(iv384, 384);
+pub const Sha512 = Sha2x64(iv512, 512);
+
+pub const Sha256T192 = Sha2x32(iv256, 256);
+
+pub const Sha512T224 = Sha2x64(iv512, 224);
+pub const Sha512T256 = Sha2x64(iv512, 256);
+
+pub const Sha512_224 = Sha2x64(truncatedSha512Iv(224), 224);
+pub const Sha512_256 = Sha2x64(truncatedSha512Iv(256), 256);
```

Comment:
Maybe just add a comment explaining what these are?

And then it's ready to be merged :)